container/go/pkg/compat/overrideConfig.go

```diff
@@ -0,0 +1,530 @@
+	Command []string
+	// Entrypoint are the new entrypoints to override entrypoint of previous layer.
+	Entrypoint []string
+	// Layer is the list of layer sha256 hashes that compose the image for which the config is written.
+	Layer []string
+	// StampInfoFile is a list of files from which to read substitutions of variables from.
+	StampInfoFile []string
+}
+
+// validate ensures that CreatedBy and Author are set.
+// based on current implementation, we expect them to never be empty.
+func (o *OverrideConfigOpts) validate() error {
+	if o.CreatedBy == "" {
+		return errors.New("CreatedBy was not specified")
+	}
+	if o.Author == "" {
+		return errors.New("Author was not specified")
+	}
+
+	return nil
+}
+
+// emptySHA256Digest returns the sha256 sum of an empty string.
+func emptySHA256Digest() (empty string) {
+	b := sha256.Sum256([]byte(""))
+	empty = hex.EncodeToString(b[:])
+	return
+}
+
+// extractValue returns the contents of a file pointed to by value if it starts with a '@'.
+func extractValue(value string) (string, error) {
+	if strings.HasPrefix(value, "@") {
+		f, err := os.Open(value[1:])
+		defer f.Close()
+		shaHash, err := ioutil.ReadAll(f)
+		if err != nil {
+			return "", errors.Wrapf(err, "failed to read content from file %s", value)
+		}
+		return string(shaHash), nil
+	}
+	return value, errors.Wrapf(nil, "unexpected value, got: %s want: @{...}", value)
+}
+
+// keyValueToMap converts an array of strings separated by '=' into a map of key-value pairs.
+func keyValueToMap(value []string) (map[string]string, error) {
+	convMap := make(map[string]string)
+	var temp []string
+	for _, kvpair := range value {
+		temp = strings.Split(kvpair, "=")
+		if len(temp) != 2 {
+			return convMap, errors.New("value in array are not of format key=value")
+		}
+		key, val := temp[0], temp[1]
+		convMap[key] = val
+	}
+	return convMap, nil
+}
+
+// Stamp provides the substitutions of variables inside {} using info in file pointed to
+// by stampInfoFile.
+func Stamp(inp string, stampInfoFile []string) (string, error) {
+	if len(stampInfoFile) == 0 || inp == "" {
+		return inp, nil
+	}
+	formatArgs := make(map[string]interface{})
+	for _, infofile := range stampInfoFile {
+		f, err := os.Open(infofile)
+		if err != nil {
+			return "", errors.Wrapf(err, "failed to open file %s", infofile)
+		}
+		defer f.Close()
+		// scanner reads line by line and discards '\n' character by default.
+		scanner := bufio.NewScanner(f)
+		var temp []string
+		for scanner.Scan() {
+			temp = strings.Split(scanner.Text(), " ")
+			key, val := temp[0], temp[1]
+			if _, ok := formatArgs[key]; ok {
+				fmt.Printf("WARNING: Duplicate value for key %s: using %s", key, val)
+			}
+			formatArgs[key] = val
+		}
+		if err = scanner.Err(); err != nil {
+			return "", errors.Wrapf(err, "failed to read line from file %s", infofile)
+		}
+	}
+
+	o := inp
+	for k, v := range formatArgs {
+		if vStr, ok := v.(string); ok {
```

Comment:
Don't need this type conversion anymore after changing formatArgs to map from string to string